### ISSUE TITLE
Explain motivation

### ISSUE THREAD

**Issue by randomstuff** [peppelinux/draft-demarco-oauth-nonce-endpoint#26]
After reading the document, it's not clear what is the use of this nonce, when you would want to request a nonce and why.

The abstract references [RFC6749](https://www.rfc-editor.org/rfc/rfc6749) but the only mention of a nonce in RFC6749 is a [non-normative example](https://www.rfc-editor.org/rfc/rfc6749#section-7.1) about an expired [OAuth-HTTP-MAC](https://datatracker.ietf.org/doc/draft-ietf-oauth-v2-http-mac/) draft.

**Comment by peppelinux** [comment]
This draft proposes creating a general-purpose endpoint to issue nonces for various purposes and audiences. This endpoint could be accessible to both public users and internal domains of the AS or other entities. This decision depends on how the nonce endpoints are integrated into the IT infrastructure.

**Comment by randomstuff** [comment]
Yes, but maybe an example of a precise use case would make it clear. Why wouldn't I use a locally generated 160 bit random number using a secure random source? Why would the AS need to encrypt (sign? MAC?) the nonce instead of generating a 160 bit random number using a secure random source? If the AS is encrypted (signed? MACed?) does anyone need to deceypt (check the signature? check this MAC?) and how?

**Comment by peppelinux** [comment]
good points that produces good clarifications, thank you. I put some sparse notes to keep the conversation going, therefore we can decide if something interesting can be further expandend and included in the current specs

- locally generated is not a microservice resource, the purpose of this specification is to create a general purpose endpoint
- encrypted to allow the AS to inspect a payload and recognize if it is the issuer. There can be also audiences and scopes within the payload, making the nonce artifact more powerful for interop purposes
- the one that decrypt the nonce is the one that shares the same private key of the nonce issuer, therefore these two entities/endpoints should be hosted in the same domain (and/or be part of the same cluster). There are cases that using appropriate audiences and scopes within the nonce payload we are able to prevent replay attacks using a local storage, without a shared  storage between many microservice. This scales better because makes a stateless layer for the nonce validation and reduces the burden of the shared storage

the purpose of this specification is to have conversation like this one and also evaluate the differences between local/shared storage, stateless/statefull approces and `jti` vs the nonce in general

**Comment by peppelinux** [comment]
another example about the scope of this individual draft is to map the intrinsic value of the nonce, and all the assumptions we have/had about it

in particular, there are some evidence around the specs, that show that the nonce could be used more than once.

From this evidence I'm considering the existence of the "Nonce Paradox" that demonstrates that the nonce, that per definition means "No more than Once" would became a **Monce** establishing that it could be used "More than once".

"The Monce Paradox", therefore, should be further explored in IETF, OpenID foundation, ISO and any other standardization body, where the constraints of uniqueness of the nonce appears to be broken.

here, for instance, I want to report the definition of nonce taken from the NIST's glossary
https://csrc.nist.gov/glossary/term/nonce

## Specifications Defining Or Referencing Nonces

- RFC 6749 (The OAuth 2.0 Authorization Framework)
- https://datatracker.ietf.org/doc/html/rfc6704
- https://datatracker.ietf.org/doc/rfc8954/
- https://datatracker.ietf.org/doc/rfc3310/ (with some limitations)
- rfc3261

## Specifications Apparently Defining Monces
- OpenID4VCI, see: https://github.com/openid/OpenID4VCI/pull/325#issuecomment-2133357644
